### PR TITLE
#42 autoload.php is enough - remove other guzzle requires

### PR DESCRIPTION
**Pull request by qbaze**
hey @Elendev,

needed this one #42 fixed asap.
autoload.php is enough, no need to specify guzzle files directly unless I don't understand something.

this one is tested and works correctly.

thanks.

### CODE REVIEW

**Comment by Elendev** [comment]
Hello @qbaze, first of all, thank you for your PR!

Are you sure about this? The PR #26 of @mveyrenc added those lines to support specifically this use case.
If something has changed with the last release of Guzzle, we need to have the `composer.json` updated to remove the support of the previous versions of Guzzle.

Or am I missing something?
Cheers!

**Comment by qbaze** [comment]
@Elendev,

I understand the need of manually requiring autoload when using your library directly not from within some framework, this is the way I use it actually.

But I don't understand the part with guzzle you described, sorry. 

Thanks!

**Comment by tm1000** [comment]
@Elendev  these commands existed well before #26 (see https://github.com/Elendev/nexus-composer-push/pull/26/files)

Infact my co-worker worked against them back in https://github.com/Elendev/nexus-composer-push/pull/16

It was originally added in https://github.com/Elendev/nexus-composer-push/commit/3bd1f7bdad1bb52b2118e513e9dc87d927a7384c

The commit itself even says

> It's a terribly nasty hack

File: src/PushCommand.php [hunk 1]
```diff
@@ -1,7 +1,18 @@
 <?php
+namespace Elendev\NexusComposerPush;
 
+if (file_exists(dirname(__DIR__) . '/vendor/autoload.php')) {
+    $loader = require_once dirname(__DIR__) . '/vendor/autoload.php';
+} elseif (file_exists(dirname(__DIR__) . '/../../autoload.php')) {
+    $loader = require_once dirname(__DIR__) . '/../../autoload.php';
+} else {
+    // error?
```
Comment (Elendev) [inline src/PushCommand.php:9]: I propose that a notice/debug is written in the console here, something like
```suggestion
  $this->getIO()
      ->write(
          '[GuzzleLoad] Failed to find Guzzle in vendors',
          true,
          IOInterface::VERY_VERBOSE
      )
    ;
```

**Review by Elendev — CHANGES_REQUESTED** [review]
I haven't been able to test all the use cases, but it seems to covers them.
When I've the time, I'll add proper testing for this.

I propose a notice when the vendors are not found, this way it might be easier to debug.

**Comment by qbaze** [comment]
@Elendev @tm1000 

I removed my previous comment as it was not correct, sorry.

Anyway requiring autoload is both necessary and enough without specifying guzzle or any specific dep files as autoloader will do the work. The only problem is finding autoload file correct location, right?

Using pure php can help,  __DIR__ variable and require_once to the rescue. I will play with this. 

Let me know what you think.

Some references below
https://github.com/composer/composer/issues/2904
https://stackoverflow.com/questions/66132997/how-to-require-a-composer-package-inside-a-plugin

Thanks!

**Comment by qbaze** [comment]
hi @Elendev,

sorry for the delay, busy time lately.

Anyway, the message is there.

Thanks!

**Review by Elendev — APPROVED** [review]

**Pull request merged**